src/RPCServer.ts

```diff
@@ -201,12 +129,53 @@ class RPCServer extends EventTarget {
       }
       never();
     }
+    this.logger.info(`Started ${this.constructor.name}`);
+  }
+  protected onTimeoutCallback?: () => void;
+  protected idGen: IdGen;
+  protected logger: Logger;
+  protected handlerMap: Map<string, RawHandlerImplementation> = new Map();
+  protected defaultTimeoutMap: Map<string, number | undefined> = new Map();
+  protected handlerTimeoutTime: number;
+  protected activeStreams: Set<PromiseCancellable<void>> = new Set();
+  protected fromError: (error: ErrorRPC<any>) => JSONValue;
+  protected filterSensitive: (key: string, value: any) => any;
+  protected middlewareFactory: MiddlewareFactory<
+    JSONRPCRequest,
+    Uint8Array,
+    Uint8Array,
+    JSONRPCResponseResult
+  >;
+  // Function to register a callback for timeout
+  public registerOnTimeoutCallback(callback: () => void) {
+    this.onTimeoutCallback = callback;
+  }
+  public constructor({
+    middlewareFactory = rpcUtilsMiddleware.defaultServerMiddlewareWrapper(),
+    handlerTimeoutTime = Infinity,
+    logger,
+    idGen = () => Promise.resolve(null),
+    fromError = rpcUtils.fromError,
+    filterSensitive = rpcUtils.filterSensitive,
+  }: {
+    middlewareFactory?: MiddlewareFactory<
+      JSONRPCRequest,
+      Uint8Array,
+      Uint8Array,
+      JSONRPCResponseResult
+    >;
+    handlerTimeoutTime?: number;
+    logger?: Logger;
+    idGen: IdGen;
```

Comment:
You forgot to make idGen optional

Reply:
879ddd5eb98a602d973e4261406b4834358903fb